app/src/main/java/org/mozilla/reference/browser/pip/PictureInPictureFeature.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.mozilla.reference.browser.pip
+
+import android.annotation.TargetApi
+import android.app.Activity
+import android.app.PictureInPictureParams
+import android.os.Build
+import androidx.fragment.app.Fragment
+import mozilla.components.browser.session.SelectionAwareSessionObserver
+import mozilla.components.browser.session.SessionManager
+import mozilla.components.support.base.feature.LifecycleAwareFeature
+import org.mozilla.reference.browser.BackHandler
+
+/**
+ * A simple implementation of Picture-in-picture mode support.
+ *
+ * @param sessionManager Session Manager for observing the selected session's fullscreen mode changes.
+ * @param activity the activity with the EngineView for calling PIP mode when required; the AndroidX Fragment
+ * doesn't support this.
+ * @param fragment the current fragment for back button handling.
+ */
+class PictureInPictureFeature(
+    sessionManager: SessionManager,
+    private val activity: Activity?,
+    private val fragment: Fragment
+) : SelectionAwareSessionObserver(sessionManager), LifecycleAwareFeature {
+
+    fun onHomePressed(): Boolean {
+        val fullScreenMode = activeSession?.fullScreenMode ?: false
+        return fullScreenMode && enterPipModeCompat()
+    }
+
+    @TargetApi(Build.VERSION_CODES.O)
+    fun enterPipModeCompat() = when (Build.VERSION.SDK_INT) {
+        in 0..Build.VERSION_CODES.N -> false
+        Build.VERSION_CODES.N, Build.VERSION_CODES.N_MR1 -> {
+            activity?.enterPictureInPictureMode()
+            true
+        }
+        else -> {
+            activity?.enterPictureInPictureMode(PictureInPictureParams.Builder().build())
+            true
+        }
```

Comment:
nit: For some reason it was a bit hard to read that. I think I was just expecting it to be the other way around:
 * If SDK greater equals O: enter with param
  * If SDK N, N_MR1: enter
  * else: do nothing

Reply:
I figured this would be the case. I was trying unnecessarily hard to get the when clause with an int to work and sacrificed readability for functional.. fixed now!